### ISSUE TITLE
Implement a "Save Event" feature

### ISSUE THREAD

**Issue by Anushka-Pote** [fed-tech/FED-Frontend#216]
## 🌟 Feature Overview
Implement a "Save Event" feature that allows users to save upcoming events they are interested in. Users will receive notifications regarding updates for these events, such as registration deadlines, schedule changes, and additional information.

## 🤔 Why this feature?
This feature enhances user engagement by allowing individuals to keep track of events that interest them. By saving events, users can stay informed about important updates and ensure they don't miss out on registration deadlines or changes. It promotes participation and helps users plan their entrepreneurial activities effectively.

## 📋 Expected Behavior
1. **Event Listing Update:**
   - Each upcoming event card should include a "Save Event" button that allows users to save the event to their profile.
  
2. **User Profile Integration:**
   - A "Saved Events" section should be added to the user profile, displaying all events the user has saved.
   - Users should be able to view event details, receive notifications for updates, and remove events from their saved list.

3. **Notifications:**
   - Users should receive notifications via email or in-app alerts for important updates related to their saved events, such as:
     - Registration deadlines
     - Event cancellations or rescheduling
     - Additional details or changes

4. **Persistence:**
   - Saved events should persist across user sessions and be accessible from any device when the user logs in.

**Comment by AnshRaj112** [comment]
This is a good suggestion; however, I believe it may not be as impactful for the society's website at this stage. Most events are announced only 2-3 weeks before registrations close or the event takes place. If we decide to scale the society beyond the college level in the future, this feature would definitely add more value. That said, there are a few exceptions, such as recruitment processes, which run for a longer duration and could benefit from this functionality.

**Comment by Anushka-Pote** [comment]
Makes sense actually, yeah. I will try to figure out some way out. But again I believe, in today's fast paced world one sees so many events and are on so many platforms that you tend to forget it, and for that integration with google calendar like via google invites is one way which people give out reminders for emails, which sometimes gets annoying as well. But the other way is via the saved list or I remember sending this to myself sort of approach.

**Comment by Anushka-Pote** [comment]
I will get the feature integrated and see how it is going to work overall, for the prototype and still if you have any other solution such as reminders or something to increase audience for the site; I think we all are open to suggestions

**Comment by AnshRaj112** [comment]
I really like the Google Calendar suggestion since it will help participants stay aware of when the event is happening. This feature could enhance event visibility and ensure better participation, so I believe it’s worth implementing.

**Comment by Anushka-Pote** [comment]
Yeah, but we are currently talking about how to engage the audience before registering, like so one can get more crowd for the events. For the Google Calendar, they will have to register first and then only they can get the regular updates and reminders. 
And.... As far as I think, then for before registration there is the Subscription for newsletter and regular updates option. 
You know the mails from unstop or other marketing platforms we constantly get for any new scheme or event or even discounts, like that will be the option then.
